test/e2e/expirable-airdrop.spec.ts

```diff
@@ -0,0 +1,188 @@
+import { then, when, given } from '@utils/bdd';
+import { toUnit } from '@utils/bn';
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { BigNumber, Wallet } from 'ethers';
+import { takeSnapshot, time, setBalance, impersonateAccount } from '@nomicfoundation/hardhat-network-helpers';
+import { MerkleTree } from 'merkletreejs';
+
+function getLeaf(address: string, amount: BigNumber) {
+  return Buffer.from(ethers.utils.solidityKeccak256(['address', 'uint256'], [address, amount]).slice(2), 'hex');
+}
+
+function createMerkleRoot(accounts: any[], amount: number) {
+  let toClaim = ethers.utils.parseUnits(String(amount));
+
+  const leaves = accounts.map((x) => {
+    let address = x.address;
+    return getLeaf(address, toClaim);
+  });
+
+  const tree = new MerkleTree(leaves, ethers.utils.keccak256, { sortPairs: true });
+  const root = tree.getHexRoot();
+
+  return [root, tree];
+}
+
+describe('Expirable airdrop', () => {
+  let nowTimestamp: number;
+  const oneMonth: number = 30 * 24 * 60 * 60; // one month in seconds
+  let expirationTimestamp: number;
+  let accounts: any[];
+  let deployer: any;
+  let root: any;
+  let tree: any;
+  let token: any;
+  let expirableAirdrop: any;
+  let airdropAmount: number = 10;
```

Comment:
I see that you are using `toUnit(airdropAmount)` in a lot of places. Maybe we could have another variable here that is `toUnit(airdropAmount)` and use that one?